### DISCUSSION TITLE
Buttons with images

### DISCUSSION THREAD

**Discussion by pej-p** [slint-ui/slint#2704] [Q&A] [answered]
I want to add an Icon to a button. The Icon should be centered and scaling automatically to the size of the button. How would I do that?

**Comment by ogoffart** [comment, marked as answer]
There is a icon property on the Button widget.  But that puts the icon on the side.

The provided Button widget cannot be customized further. So what you can do is to have a Button without content and put the Image over it,  or write your own Button.

[Example](https://slint-ui.com/releases/1.0.2/editor/?snippet=import+%7B+Button%2C+VerticalBox+%7D+from+%22std-widgets.slint%22%3B%0A%0Aexport+component+ButtonWithImage+%7B%0A++++in+property+image+%3C%3D%3E+img.source%3B%0A++++callback+clicked+%3C%3D%3E+btn.clicked%3B%0A++++btn+%3A%3D+Button+%7B%0A++++++++width%3A100%25%3B%0A++++++++height%3A+100%25%3B%0A++++%7D%0A++++lay+%3A%3D+HorizontalLayout+%7B%0A++++++++padding%3A+5px%3B%0A++++++++img+%3A%3D+Image+%7B+%0A++++++++++++preferred-height%3A+image.height+*+1px%3B%0A++++++++++++preferred-width%3A+image.width+*+1px%3B++++++++%0A++++++++%7D++%0A++++%7D%0A%7D%0A%0Aexport+component+Demo+%7B%0A++++VerticalBox+%7B%0A++++++++alignment%3A+start%3B%0A++++++++Text+%7B%0A++++++++++++text%3A+%22Hello+World%21%22%3B%0A++++++++++++font-size%3A+24px%3B%0A++++++++++++horizontal-alignment%3A+center%3B%0A++++++++%7D%0A++++++++ButtonWithImage+%7B%0A++++++++++++image%3A+%40image-url%28%22https%3A%2F%2Fslint-ui.com%2Flogo%2Fslint-logo-full-light.svg%22%29%3B%0A++++++++%7D%0A++++++%0A++++%7D%0A%7D%0A)

**Reply by pej-p** [reply]
Thanks, works like a charm! Is there also an example for changing the content on click (Like the content of the window or the icon of the button)?

**Reply by tronical** [reply]
How about something like [this](https://slint-ui.com/releases/1.0.2/editor/?snippet=import+%7B+Button%2C+VerticalBox+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++property+%3Cbool%3E+toggle%3B%0A++++Button+%7B%0A++++++++text%3A+%22Hello%22%3B%0A++++++++clicked+%3D%3E+%7B%0A++++++++++++toggle+%3D+%21toggle%3B%0A++++++++++++if+%28toggle%29+%7B%0A++++++++++++++++self.icon+%3D+%40image-url%28%22https%3A%2F%2Fslint-ui.com%2Flogo%2Fslint-logo-full-light.svg%22%29%3B%0A++++++++++++%7D+else+%7B%0A++++++++++++++++self.icon+%3D+%40image-url%28%22https%3A%2F%2Fslint-ui.com%2Flogo%2Fslint-logo-square-light.svg%22%29%3B%0A++++++++++++%7D%0A++++++++%7D%0A++++%7D+++%0A%7D%0A) ? :)
